### ISSUE TITLE
Sparse vs Dense Activation in MoE Implementation

### ISSUE THREAD

**Issue by grisaiakaziki** [sitzikbs/Neural-Experts#1]
Congratulations on your paper acceptance! 🎉 It's exciting to see impactful work being recognized.  

I have a question about the implementation details of the sparse MoE activation described in Fig.1(c) of the paper. In the code snippet:  

sdf_term = self.sdf_loss(non_manifold_pred, nonmnfld_sdf_gt, nonmnfld_q)

it appears that the SDF loss is computed densely across sampled points (`non_manifold_pred`), which seems contradictory to the paper's sparse activation.  

Could you help clarify:  
1. Is the MoE indeed using **dense activation during training** and switching to sparse activation only during inference?  
2. If so, what is the motivation behind this design choice?

I appreciate your time and look forward to learning from your insights!

**Comment by sitzikbs** [comment]
Hi, 
Thank you for your interest in our work. 
1+2. Training is Dense. That is, it needs to compute the loss for all experts. This is in line with Hinton's original formulation for the loss used for training Mixture of Expert networks.
 At inference, it can and should be sparse (i.e. only the relevant expert is evaluated), so essentially the manager network makes it's prediction of which expert to use per input coordinate and then only the relevant expert is evaluated. 

The rationale behind this design choice for the training loss is detailed in Hinton's paper but the gist of it is that this loss formulation is what encourages each expert to specialize in its subspace of the input domain(and  not average). 
I hope this clarifies things. 
Itzik

**Comment by grisaiakaziki** [comment]
Thank you so much for your prompt and helpful response—I really appreciate your time and assistance!
